src/libraries/Native/Unix/System.Native/CMakeLists.txt

```diff
@@ -70,6 +70,10 @@ if (GEN_SHARED_LIB)
     )
 
     install_with_stripped_symbols (System.Native PROGRAMS .)
+
+    if (HAVE_GETADDRINFO_A)
+        target_link_libraries(System.Native anl)
```

Comment:
This should be done in extra_libs.cmake, otherwise noone else will know that there is an extra lib involved

Reply:
Also, since it is a CMAKE_REQUIRED_LIBRARIES for Linux, it may be more reliable to check if (CLR_CMAKE_TARGET_LINUX). 

Alternatively, everything that needs to link this will also need to know about `HAVE_GETADDRINFO_A`. 
Thus if this is required on Linux and Linux only, checking for Linux would be simpler.

---
In reply to: [545597019](https://github.com/dotnet/runtime/pull/34633#discussion_r545597019) [](ancestors = 545597019)